### ISSUE TITLE
Configuring Key Bindings and Fuzzy Auto-Completion in Ubuntu 22.04

### ISSUE THREAD

**Issue by cvladan** [junegunn/fzf#3395]
It's simply astonishing how challenging it is to get this working on the most popular Linux distribution, especially given that the documentation is far from clear and straightforward. That's why I've decided to write this up.

To be specific, my system is Ubuntu 22.04, my shell of choice is bash, and I'm on Oracle Cloud.

Everything should be achieved easily by running:

```sh
sudo apt -y install fzf
```
But alas!

### 1. Not enabled by default 

It's essential to know that on Debian-based distributions like Ubuntu is, key bindings (Ctrl+T / Ctrl+R / Alt+C) and fuzzy auto-completion are not enabled by default. You must add them in your `~/.bashrc`.

```bash
echo '[ -f /usr/share/doc/fzf/examples/key-bindings.bash ] && source /usr/share/doc/fzf/examples/key-bindings.bash' >> ~/.bashrc
echo '[ -f /usr/share/bash-completion/completions/fzf ] && source /usr/share/bash-completion/completions/fzf' >> ~/.bashrc
```

### 2. Files are Not Downloaded + Latest Version Has Fatal Error

Everything would have been simple if those files were present - but they aren't. Not only are they missing, but the current file `key-bindings.bash` is also faulty as it is responsible for throwing an error `unknown option: --scheme=history` on `Ctrl-T`. So you'll need an older version (commit: 6dcf5c3).

You'll have to manually download them and store them where they're supposed to be:

```bash
if [[ ! -f /usr/share/doc/fzf/examples/completion.bash ]]; then
  sudo curl https://raw.githubusercontent.com/junegunn/fzf/master/shell/completion.bash -o /usr/share/doc/fzf/examples/completion.bash
  curl https://raw.githubusercontent.com/junegunn/fzf/6dcf5c3d7d6c321b17e6a5673f1533d6e8350462/shell/key-bindings.bash -o /usr/share/doc/fzf/examples/key-bindings.bash
fi
```
### 3. Just One More Re-login Cycle

Lastly, to get all of this sourced into the system, it's easiest for me to open a new session and test it.

---

### It's Slightly Easier for Zsh

For the .zsh shell, just alter the file extensions everywhere to .zsh and there's no older version required - I guess this, but I didn't actually test it:

```zsh
if [[ ! -f /usr/share/doc/fzf/examples/completion.zsh ]]; then
  sudo curl https://raw.githubusercontent.com/junegunn/fzf/master/shell/completion.zsh -o /usr/share/doc/fzf/examples/completion.zsh
  sudo curl https://raw.githubusercontent.com/junegunn/fzf/master/shell/key-bindings.zsh -o /usr/share/doc/fzf/examples/key-bindings.zsh
fi
```

### In summary, a serious nuisance

Isn't there a way to make this simpler?

**Comment by junegunn** [comment]
You might want to contact the maintainers of the package instead (I'm not one of them).

> but the current file `key-bindings.bash` is also faulty

There is no problem with the file. You're getting the error because you're using it with an old version of fzf binary which doesn't have the `--scheme` option.

**Comment by cvladan** [comment]
Possible, but that is the fzf version that gets installed with `apt install fzf`, which is why I wrote these instructions.

**Comment by junegunn** [comment]
I know, that should be an old version and you can't use the latest scripts on this repo with it.

**Comment by cvladan** [comment]
In any case, thousands of users want to use fzf key bindings on Ubuntu, and this is the easiest and possibly the only way to do it.

This is also the recommended method for installing fzf on Linux, and without these instructions that I have written down, it is NOT possible to use key bindings on Ubuntu.

I also think that this should be written somewhere in the installation instructions, but in the meantime I have documented it here for everyone - as the amount of effort required to set this up really surprised me.

It's not about where to report a bug, but the idea is that people can use this feature, and they should have the instructions written in one place, not scattered across multiple issues in multiple repos.

I suppose we don't want to say "you can't use the fzf key bindings on Ubuntu"?

**Comment by junegunn** [comment]
You're suggesting that the Ubuntu package needs fixing, but posting it here won't trigger any action from the actual maintainers.

I have used Ubuntu in the past, and I have always struggled with the outdated packages it provides. You end up coming to terms with the limitation of the default package manager and looking for other ways to install newer versions of the package. (e.g. building from source, using alternative package managers such as Homebrew, etc.)

> This is also the recommended method for installing fzf on Linux, and without these instructions that I have written down, it is NOT possible to use key bindings on Ubuntu.

There are other ways to install fzf on Linux and they're mentioned in https://github.com/junegunn/fzf#installation. If the package from the default package manager is broken or if you need a different (newer) version than it provides, you will have to use another way.

**Comment by mcepl** [comment]
@cvladan Please, file your bug report at https://bugs.launchpad.net/ubuntu/+filebug where it belongs.

**Comment by junegunn** [comment]
I'm going to close this because it's not a bug in fzf, it's a bug in the Ubuntu package, and/or it's caused by a misconfiguration.

As of fzf 0.48.0, the scripts are embedded in the binary. So the users are less likely to have incompatibility problems.

See #3675
